doc/user_guide.md

```diff
@@ -0,0 +1,2082 @@
+
+![img/filter-04.png](img/filter-04.png) 
+
+`Fritz` automatically prepends these stages to all user-defined filters. However, when constructing/debugging
+filters in Compass, the users must take care of that -- all the examples below come with the upstream stages prepended.
+
+
+#### Limitations
+
+`$lookup`, `$unionWith`, `$out`, and `$merge` stages are not allowed in the user-defined part of the filters.
+
+### Filter examples
+
+Let us explore some examples! The provided JSON files contain prepended upstream stages, but they are not shown in the
+text below for brevity.
+
+#### Simple filter
+
+Let us start with a very simplistic example:
+
+[fritz_filter_101.json](data/filter_examples/fritz_filter_101.json)
+
+The first user-defined [`$match`](https://docs.mongodb.com/manual/reference/operator/aggregation/match/) 
+stage selects alerts with deep-learning-based real-bogus scores (`candidate.drb`) greater than 0.9999999 
+that don't have any matches with the `Gaia_DR2` catalog 
+(the zeroth element of the corresponding array does not exist)
+
+```json
+{
+  "$match": {
+    "candidate.drb": {
+      "$gt": 0.9999999
+    },
+    "cross_matches.Gaia_DR2.0": {
+      "$exists": false
+    }
+  }
+}
+```
+
+In case an alert passes the first stage, the output stage adds the annotations dictionary:
+
+```json
+{
+  "$project": {
+    "_id": 0,
+    "candid": 1,
+    "objectId": 1,
+    "annotations.author": "dd",
+    "annotations.mean_rb": {
+      "$avg": "$prv_candidates.rb"
+    }
+  }
+}
+```
+
+`candid` and `objectId` are kept here for the debugging purposes, to see which alerts pass your filter.
+
+The output of the filter will look something like this:
+
+![img/filter-05.png](img/filter-05.png)
+
+This alert will be posted to the candidates page with these annotations. 
+
+#### CLU filter
+
+Now that we've looked at a basic example, let us explore a real-life example and build a filter for 
+the [Census of the Local Universe](https://ui.adsabs.harvard.edu/abs/2020arXiv200409029D/abstract) program.
+
+As a reference, we will use the filter definition (as of July 10, 2020) 
+from the GROWTH marshal translated into `python` code (copy-pasted here with no alterations):
+
+```python
+def clu_filter(current_observation):
+    filteron = False
+    annotations = {}
+    calccount = 10000
+    bright = False
+    nopointunderneath = True
+    mover = True
+    real = False
+    tdiff = (-99.0)
+    magdiff = (-99)
+    riserate = (-99)
+    decayrate = (-99)
+    hostgr = (-99)
+    hostri = (-99)
+    positivesubtraction = False
+    brightstar = False
+    highlum = False
+    deltajd = 0
+    prevcandidates = current_observation['prv_candidates']
+    m_now = current_observation['candidate']['magpsf']
+    m_app = current_observation['candidate']['magap']
+    t_now = current_observation['candidate']['jd']
+    fid_now = current_observation['candidate']['fid']
```

Comment:
Is there a standard reference for what all these parameters are? What range is expected?

Reply:
Yes, it lives [here](https://github.com/ZwickyTransientFacility/ztf-avro-alert). Added an explicit mention in the text.